servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/TransportObserverInitializer.java

```diff
@@ -92,4 +139,32 @@ public void flush(final ChannelHandlerContext ctx) {
             ctx.flush();
         }
     }
+
+    @Sharable
+    static final class SecurityHandshakeObserverHandler extends ChannelDuplexHandler {
+
+        static final SecurityHandshakeObserverHandler INSTANCE = new SecurityHandshakeObserverHandler();
+
+        private SecurityHandshakeObserverHandler() {
+            // Singleton
+        }
+
+        @Override
+        public void userEventTriggered(final ChannelHandlerContext ctx, final Object evt) {
+            if (evt instanceof SslHandshakeCompletionEvent) {
+                final Channel channel = ctx.channel();
+                final SecurityHandshakeObserver observer = securityHandshakeObserver(channel);
+                assert observer != null;
+                final SSLSession sslSession = extractSslSession(ctx.pipeline(), (SslHandshakeCompletionEvent) evt,
```

Comment:
We are already intercepting this event and extract SSL session in `DefaultNettyConnection`, can we do these callbacks there instead of adding another handler?

Reply:
There is `H2ParentConnectionContext` that also performs a similar logic, but this class is from `servicetalk-http-netty`.
The goal to have `SecurityHandshakeObserverHandler` is to encapsulate everything related to transport observer in `servicetalk-transport-netty-internal`. For users who are interested in observability, it looks like small and acceptable overhead.

Reply:
Moved this logic to `NettyPipelineSslUtils.extractSslSession` instead: https://github.com/apple/servicetalk/pull/1099/commits/e118cd1c3dbf862f5d4a4e8fe1467bffdfdeb7db